### PR TITLE
Introduce outline of new traits

### PR DESCRIPTION
**Pull request by lukemorton**
This is based on previous conversations between myself and various colleagues.

The aim here will be to introduce both needs and concrete expectations so that people know exactly what we're looking for and can easily be measured. At the moment I've only added needs to trigger a conversation around these traits.

Currently taking input on:

 - Other traits that we think would be valuable to introduce now
 - Suggestions on expectations for traits
 - Updates to needs for the traits listed here

### CODE REVIEW

File: roles/engineer.md [hunk 1]
```diff
@@ -34,3 +64,9 @@ We expect someone in the role:
 - to need less than 1 day per week of Technical Director involvement in their deliveries
 - to be mentoring middleweight developers on their team within 2 months
 - to be improving our practices across the business within 3 months
+
+### +Mentor
+
+We need someone:
+
+- to be helping colleagues and company friends develop their skills
```
Comment (craigjbass) [inline roles/engineer.md]: This is very broad

**Comment by lukemorton** [comment]
It might be that +Lead and +Mentorship are too similar and we only go ahead with one for now.

Previous thinking was that +Lead trait has already started appearing amongst teams and is typically a more experienced member of the team who helps push teams forward, provides more of a scrum/flow master role, pulls everyones socks up in terms of code standards, etc.

+Mentor is more about helping those in your team learn new skills or helping people in the wider company or even outside the company.

### +Mentor

We need someone:

 - to provide a mentorship role to a colleague
 - to encourage learning across the company

File: roles/engineer.md [hunk 2]
```diff
@@ -20,6 +20,36 @@ We expect someone in the role:
 
 ## Engineer traits
 
+### +Advocate
+
+We need someone:
+
+ - to represent the company by speaking in public
+ - to represent the company on social media
+ - to represent the companies hiring interests publicly
```
Comment (chrisblackburn) [inline roles/engineer.md]: company's

**Comment by craigjbass** [comment]
@lukemorton is the outcome to have a broad approach to each role, or a very narrow approach

I would say +Coach would make sense instead of +Lead, but then this would not predicate a tech background. 

If the tech knowledge is split out this could then be +Mentor but then this predicates some mentoring ability in one or more particular stacks. This might be a large barrier to entry if we want to go down the Polyglot approach and have lots of stacks. Potentially each +Mentor could be in a particular stack, but then to have a role around a stack would potentially become a business barrier to entry to just picking up a new stack as there is already inertia around other stacks.

+Architecture is interesting as this should encapsulate transferable knowledge, and be stack agnostic.

I would say I can mentor on generalised knowledge that can apply ATDD to Ruby, PHP, Java, Kotlin but it would be a joke for me to attempt to mentor someone on Rails or Spree at this time.

**Comment by lukemorton** [comment]
I would say all traits will be agnostic to a particular technology.

**Comment by craigjbass** [comment]
Do you expect everyone to get +Mentor if they are capable of mentoring

**Comment by lukemorton** [comment]
I think we will layout expectations as per the Engineer role and existing traits which will be quantifiable to the point we can say yes this person exhibits this trait or not.

If everyone fulfils the criteria of the +Mentor then they are indeed +Mentor.

**Comment by craigjbass** [comment]
Perhaps a narrow approach would be highly beneficial

**Comment by lukemorton** [comment]
Okay I feel we've got a nice mix of traits at the moment. I've left +Lead in for now, however it looks more likely that more specific traits will end up superseding the need for +Lead.

Merging this PR in. Next couple of PRs will handle +Lead and expectations for traits.

**Pull request merged**